### PR TITLE
Add promptflow-core test

### PR DESCRIPTION
**Pull request by brynn-code**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by wangchao1230 — DISMISSED** [review]

File: src/promptflow-core/tests/utils.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from pathlib import Path
+
+TEST_CONFIG_ROOT = Path(__file__).parent.parent.parent / "promptflow" / "tests" / "test_configs"
+FLOW_ROOT = TEST_CONFIG_ROOT / "flows"
+EAGER_FLOW_ROOT = TEST_CONFIG_ROOT / "eager_flows"
+
+
+def get_flow_folder(folder_name, root: str = FLOW_ROOT) -> Path:
```
Comment (crazygao) [inline src/promptflow-core/tests/utils.py:8]: Move this as conftest.py?

Comment (crazygao) [reply src/promptflow-core/tests/utils.py:8]: remove the __init__ in this layer

Comment (brynn-code) [reply src/promptflow-core/tests/utils.py]: Left init there, renamed conftest

**Review by lumoslnt — DISMISSED** [review]

**Review by wangchao1230 — DISMISSED** [review]

File: scripts/check_enforcer/check_enforcer.py [hunk 2]
```diff
@@ -57,6 +57,11 @@
         "scripts/building/**",
         ".github/workflows/promptflow-sdk-cli-azure-test-pull-request.yml",
     ],
+    "promptflow-core-test [Pure]": [
```
Comment (crazygao) [inline scripts/check_enforcer/check_enforcer.py:60]: No [Pure]

**Review by wangchao1230 — APPROVED** [review]

**Review by crazygao — APPROVED** [review]

**Pull request merged**